Github Deployments i1: Add GitHub deployments section

vykes-mac opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/5247

## Proposed Changes

* Renders github deployments section at `/github-deployments/:siteId`
* Enabled and reuse the `github-integration-i1` flag on dev

![image](https://github.com/Automattic/wp-calypso/assets/47489215/34f34166-6d6e-4797-b54b-77bf4d429746)


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* go to `/github-deployments/atomic_site` verify the github deployments page
* go to `/github-deployments/simple_site` verify redirected to `/home`

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

vykes-mac committed:
render github deployment section at `/github-deployments/:siteId`

vykes-mac committed:
enable github-integration flag on dev

Aurorum commented on client/my-sites/github-deployments/controller.tsx:
```diff
@@ -0,0 +1,34 @@
+export const githubDeployments: Callback = ( context, next ) => {
+	context.primary = (
+		<>
+			<PageViewTracker path="/github-deployments/:site" title="Github Deployments" delay={ 500 } />
+			<GithubDeployments />
```
Maybe just me, but I think it'd be preferable to capitalise Github - ie. `<GitHubDeployments />`

I feel that's particularly important for strings shown on the front-end in the other file and on the sidebar.

vykes-mac replied:
You are correct, It should be capitalised.

Aurorum commented on client/my-sites/github-deployments/index.ts:
```diff
@@ -0,0 +1,18 @@
+
+	page(
+		'/github-deployments/:siteId',
+		siteSelection,
```
You'll probably need to add the new section here so that it's translatable: https://github.com/Automattic/wp-calypso/blob/b783cbb24dbbde9e9168893b2207da575c0b57cc/client/my-sites/sites/index.jsx#L119-L156

At the moment, it just shows this.
<img width="500" alt="Screenshot 2024-01-31 at 20 03 00" src="https://github.com/Automattic/wp-calypso/assets/43215253/04a3b152-a97e-4834-86a5-f5089c60d3d0">

Aurorum commented on client/my-sites/github-deployments/main.tsx:
```diff
@@ -0,0 +1,39 @@
+						components: {
+							learnMoreLink: (
+								<InlineSupportLink
+									key="learnMore"
```
Just out of interest, is the key necessary?

vykes-mac replied:
The key can be removed, oversight from copying.

Aurorum commented on client/my-sites/github-deployments/main.tsx:
```diff
@@ -0,0 +1,39 @@
+//import './style.scss';
+
+export function GithubDeployments() {
+	const { __ } = useI18n();
```
This is more a nit than anything really - but might be better to just stick with `translate` or `useI18n()` rather than mixing both? (unless I'm missing something, of course!)

vykes-mac committed:
add github deployments to Sites translatable strings

vykes-mac committed:
- Capitalise GitHub string and component
- Use translate function to translate strings in `<GitHubDeployments/>` component

vykes-mac commented:
Thanks @Aurorum for the review, I've addressed your comments

zaguiini commented on client/my-sites/github-deployments/controller.tsx:
```diff
@@ -0,0 +1,34 @@
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+
+	if ( ! isEnabled( 'github-integration-i1' ) || ! isAtomicSite( state, siteId ) ) {
```
Let's move forward with that in this PR, but create a follow up to transfer the site to Atomic if eligible, a la `/hosting-config/%s`.

vykes-mac replied:
Yeah we probably gonna show an upsell or something

zaguiini reviewed (approved):
Works well 👍

Aurorum commented on client/my-sites/github-deployments/controller.tsx:
```diff
@@ -0,0 +1,34 @@
+export const githubDeployments: Callback = ( context, next ) => {
+	context.primary = (
+		<>
+			<PageViewTracker path="/github-deployments/:site" title="Github Deployments" delay={ 500 } />
```
```suggestion
			<PageViewTracker path="/github-deployments/:site" title="GitHub Deployments" delay={ 500 } />
```

Aurorum commented on client/my-sites/sidebar/static-data/fallback-menu.js:
```diff
@@ -517,6 +518,17 @@ export default function buildFallbackResponse( {
+							{
+								parent: 'tools.php',
+								slug: 'tools-github-deployments',
+								title: translate( 'Github Deployments' ),
```
```suggestion
								title: translate( 'GitHub Deployments' ),
```

Aurorum reviewed (approved):
Thanks for addressing my comments! Just two more strings to update but otherwise I also think this looks great :)

vykes-mac committed:
fix strings

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11151517

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11151517).
